README


This archive contains the distribution Dist-Zilla-PluginBundle-Author-RWSTAUNER,
version 4.203:

  RWSTAUNER's Dist::Zilla config

This software is copyright (c) 2010 by Randy Stauner.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.025.



Changes
Revision history for Dist-Zilla-PluginBundle-Author-RWSTAUNER

4.203     2014-12-11T23:47:53Z

  - Trade Bugtracker and Repository for AutoMetaResources.
  - Put PkgVersion before other FileMunger plugins
  - Add local, perl5, and fatlib as no_index directories

4.202     2014-05-05T03:31:07Z

  - Bump prereq on @TestingMania to use Test::NoTabs instead of NoTabsTest.
  - Change README minter template from pod to markdown.

4.201     2014-02-01T16:40:02Z

  - Bump Dist::Zilla and Pod::Weaver prereqs for Encoding updates.

4.200     2014-02-01T16:19:23Z

  - Change Pod::Weaver::Plugin '-Encoding' for the new core '-SingleEncoding'.
    Thanks, Sergey Romanov!
  - Change generated README in repo root back to markdown to get metacpan urls.

4.102     2013-07-09T23:28:02Z

  - Add 'open_source' option which can be disabled for darkpan projects
    to exclude various cpan/meta/repo/xt/change log plugins.

4.101     2013-04-10T05:42:18Z

  [Test Fixes]
  - Fix unintentional failure to stringify dir arg to Git::Wrapper in tests
    to avoid test report failures with Git::Wrapper 0.028_96 dev release

4.100     2013-04-06T03:22:21Z

  - Add MintingProfile and use PluginBundle::Easy to generate mint files
  - Bump Pod::Weaver::Section::Support prereq to 1.005 and configure it
    to only link to metacpan (since that links to everything else).

4.000     2013-03-24T05:25:29Z

  - Use BundleInspector plugin to determine prereq info and generate
    "roughly equivalent" ini configs in pod.
  - Include plugin version requirements in plugin specs as ':version'
    instead of 'use' statements (and get rid of the use statements).
  - Revert to basic Readme in dist
  - Generate README.pod (instead of markdown) in repo root for github
    (ReadmeAnyFromPod instead of ReadmeMarkdownFromPod + CopyReadmeFromBuild)
  - Remove 'disable_tests' and 'skip_prereqs' in favor of role-based
    '-remove' and 'AutoPrereqs.skip'
  - Add Tests::ChangesHasContent, CheckMetaResources, CheckPrereqsIndexed
  - Collect contributors from git and list in meta and pod
  - Collect =class_method pod commands

3.203     2011-12-10T00:02:58Z

  - Consume PluginRemover role to get '-remove' magically

3.202     2011-11-19T18:25:57Z

  - Put 'Changes' back into @Git's allow_dirty
  - Reorder plugins so ConfirmRelease is last BeforeRelease

3.201     2011-11-18T04:20:13Z

  - Use ReadmeMarkdownFromPod and CopyReadmeFromBuild to automate
    a more useful github readme
  - Load @Git bundle earlier (before releaser)

3.200     2011-10-19T23:49:56Z

  - Move some logic out into Config::MVP::Slicer
    and consume Dist::Zilla::Role::PluginBundle::Config::Slicer
  - Update tests

3.108     2011-10-03T13:59:32Z

  [Prereqs]
  - Pod::Weaver::Plugin::Encoding
  - By default just use MakeMaker; set 'builder' to 'mb' or 'both'
    for ModuleBuild and/or DualBuilders
  - Bump @TestingMania required version to 0.14
  - Don't require OurPkgVersion

  [Fixes]
  - s/CompileTests/Test::Compile/g
  - Stop pruning dist.ini; it should be a familiar file for people
  - Fix regexp to actually prune cover_db/

3.107     2011-08-16T23:01:28Z

  [Changes]
  - Use 'placeholder_comments' config attribute to decide how
    $VERSION and $AUTHORITY should work (default to false).

  [Prereqs]
  - Require Prepender 1.112280 to skip prepending to tests
  - Require rather than recommend DZP::Run

3.106     2011-08-10T02:55:03Z

  [Changes]
  - use place holder comments for VERSION and AUTHORITY
  - Place Acknowledgements above Authors pod (and add stop word)

  [Prereqs]
  - Trade PkgVersion for OurPkgVersion
  - Require latest version of Authority
  - Use DZP Test::PodSpelling instead of DZP PodSpellingTests
  - Recommend Dist::Zilla::Plugin::Run (used by some projects)

3.105     2011-07-14T23:41:38Z

  - Update metadata

3.104     2011-07-05T23:36:30Z

  - Don't automatically require Test::More 0.96
  - Don't include Devel::Cover's cover_db/ in dist
  - Dump plugins to log if $ENV{DZIL_BUNDLE_DEBUG}
  - Fix regexp to match class by "Plugin" or "PluginBundle"

3.103     2011-06-22T03:07:14Z

  - Fix name of recommended plugin (PodSpellingTests not SpellingTests)

3.102     2011-06-22T03:01:05Z

  [Prereqs]
  - Dist::Zilla::Plugin::PodSpellingTests no longer a hard requirement

  [Fixes]
  - Don't overwrite CompileTests.fake_home with default if it's set in config
  - Ignore any number of Bundle prefixes from plugin names
    (to allow dynamic configuration of plugins added by bundles)
  - Improvements to tests

3.101     2011-06-16T02:52:05Z

  - Require Bootstrap::lib plugin so that it's available
  - Allow cpanm to choose best place to install (install_command)
  - Remove qw(anno forum kwalitee) from SUPPORT websites
  - Remove Test::Pod::No404s since most links are generated

3.100     2011-05-10T03:16:24Z

  - Use CheckExtraTests instead of ExtraTests to keep xt/ directory
  - Change date format (in Changes file) to W3CDTF for unambiguous dates
  - Match skip_plugins against full name or package (rather than suffix)
  - Allow blank 'releaser' attribute to skip default releaser plugin

  [Prereqs]
  - Bump required Dist::Zilla to 4.200005
  - Bump required @TestingMania to 0.010 (bug fix)

3.000     2011-04-14T05:24:45Z

  - Rework configure() to modify actual 'plugins' attribute
    so that add_plugins() and add_bundle() can be used

  [Add Dist::Zilla Plugins]
  - use InstallRelease with 'install_command' attribute
  - use @TestingMania bundle
  - use @Git bundle (instead of individual plugins)

  - Explicitly prune some non-release files
  - Generate a MANIFEST.SKIP file to satisfy tests/historical expectations

2.001     2011-04-13T02:25:30Z

  - "use" recently required modules (*::Test::Pod::*)
  - test script fix

2.000     2011-04-12T15:32:00Z

  - Rename module into Author:: namespace

  - Use the @Git bundle by default

  - Replace deprecated Dist::Zilla::Plugin::PodLinkTests with
    Dist::Zilla::Plugin::Test::Pod::LinkCheck and

Makefile.PL
#
# This file is part of Dist-Zilla-PluginBundle-Author-RWSTAUNER
#
# This software is copyright (c) 2010 by Randy Stauner.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.025.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

use File::ShareDir::Install;
$File::ShareDir::Install::INCLUDE_DOTFILES = 1;
$File::ShareDir::Install::INCLUDE_DOTDIRS = 1;
install_share dist => "share";
install_share module => "Dist::Zilla::MintingProfile::Author::RWSTAUNER", "share\/profiles";


my %WriteMakefileArgs = (
  "ABSTRACT" => "RWSTAUNER's Dist::Zilla config",
  "AUTHOR" => "Randy Stauner <rwstauner\@cpan.org>",
  "BUILD_REQUIRES" => {
    "Module::Build" => "0.3601"
  },
  "CONFIGURE_REQUIRES" => {
    "File::ShareDir::Install" => "0.06",
    "Module::Build" => "0.3601"
  },
  "DISTNAME" => "Dist-Zilla-PluginBundle-Author-RWSTAUNER",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::PluginBundle::Author::RWSTAUNER",
  "PREREQ_PM" => {
    "Data::Section" => 0,
    "Dist::Zilla" => 5,
    "Dist::Zilla::Plugin::Authority" => "1.005",
    "Dist::Zilla::Plugin::AutoMetaResources" => 0,
    "Dist::Zilla::Plugin::AutoPrereqs" => 0,
    "Dist::Zilla::Plugin::Bootstrap::lib" => 0,
    "Dist::Zilla::Plugin::CheckChangesHasContent" => "0.006",
    "Dist::Zilla::Plugin::CheckExtraTests" => 0,
    "Dist::Zilla::Plugin::CheckMetaResources" => 0,
    "Dist::Zilla::Plugin::CheckPrereqsIndexed" => 0,
    "Dist::Zilla::Plugin::ConfirmRelease" => 0,
    "Dist::Zilla::Plugin::ContributorsFromGit" => "0.005",
    "Dist::Zilla::Plugin::ExecDir" => 0,
    "Dist::Zilla::Plugin::GatherDir" => 0,
    "Dist::Zilla::Plugin::GenerateFile" => 0,
    "Dist::Zilla::Plugin::Git::Describe" => 0,
    "Dist::Zilla::Plugin::Git::Init" => 0,
    "Dist::Zilla::Plugin::Git::NextVersion" => 0,
    "Dist::Zilla::Plugin::GithubMeta" => "0.10",
    "Dist::Zilla::Plugin::InstallRelease" => "0.006",
    "Dist::Zilla::Plugin::License" => 0,
    "Dist::Zilla::Plugin::MakeMaker" => 0,
    "Dist::Zilla::Plugin::Manifest" => 0,
    "Dist::Zilla::Plugin::ManifestSkip" => 0,
    "Dist::Zilla::Plugin::MetaConfig" => 0,
    "Dist::Zilla::Plugin::MetaJSON" => 0,
    "Dist::Zilla::Plugin::MetaNoIndex" => "1.10113",
    "Dist::Zilla::Plugin::MetaProvides::Package" => "1.14000001",
    "Dist::Zilla::Plugin::MetaYAML" => 0,
    "Dist::Zilla::Plugin::MinimumPerl" => "1.003",
    "Dist::Zilla::Plugin::NextRelease" => 0,
    "Dist::Zilla::Plugin::OurPkgVersion" => "0.002",
    "Dist::Zilla::Plugin::PkgVersion" => 0,
    "Dist::Zilla::Plugin::PodWeaver" => 0,
    "Dist::Zilla::Plugin::Prepender" => "1.112280",
    "Dist::Zilla::Plugin::PruneCruft" => 0,
    "Dist::Zilla::Plugin::PruneFiles" => 0,
    "Dist::Zilla::Plugin::Readme" => 0,
    "Dist::Zilla::Plugin::ReadmeAnyFromPod" => "0.120120",
    "Dist::Zilla::Plugin::Run" => "0.014",
    "Dist::Zilla::Plugin::Run::AfterMint" => 0,
    "Dist::Zilla::Plugin::ShareDir" => 0,
    "Dist::Zilla::Plugin::TemplateModule" => 0,
    "Dist::Zilla::Plugin::Test::ChangesHasContent" => "0.006",
    "Dist::Zilla::Plugin::Test::PodSpelling" => 0,
    "Dist::Zilla::Plugin::Test::ReportPrereqs" => "0.004",
    "Dist::Zilla::Plugin::TestRelease" => 0,
    "Dist::Zilla::Plugin::UploadToCPAN" => 0,
    "Dist::Zilla::PluginBundle::Git" => "2.004",
    "Dist::Zilla::PluginBundle::TestingMania" => "0.22",
    "Dist::Zilla::Role::MintingProfile::ShareDir" => 0,
    "Dist::Zilla::Role::PluginBundle::Config::Slicer" => 0,
    "Dist::Zilla::Role::PluginBundle::Easy" => 0,
    "Dist::Zilla::Role::PluginBundle::PluginRemover" => 0,
    "Dist::Zilla::Stash::PodWeaver" => "1.005",
    "Git::Wrapper" => 0,
    "List::Util" => 0,
    "Moose" => 0,
    "Moose::Util::TypeConstraints" => "1.01",
    "MooseX::AttributeShortcuts" => 0,
    "Pod::Elemental::Transformer::List" => 0,
    "Pod::Markdown" => "2.000",
    "Pod::Weaver" => 4,
    "Pod::Weaver::Config::Assembler" => 0,
    "Pod::Weaver::Plugin::SingleEncoding" => 0,
    "Pod::Weaver::Plugin::StopWords" => "1.005",
    "Pod::Weaver::Plugin::Transformer" => 0,
    "Pod::Weaver::Plugin::WikiDoc" => 0,
    "Pod::Weaver::PluginBundle::CorePrep" => 0,
    "Pod::Weaver::Section::Authors" => 0,
    "Pod::Weaver::Section::Collect" => 0,
    "Pod::Weaver::Section::Contributors" => 0,
    "Pod::Weaver::Section::Generic" => 0,
    "Pod::Weaver::Section::Leftovers" => 0,
    "Pod::Weaver::Section::Legal" => 0,
    "Pod::Weaver::Section::Name" => 0,
    "Pod::Weaver::Section::Region" => 0,
    "Pod::Weaver::Section::Support" => "1.005",
    "Pod::Weaver::Section::Version" => 0,
    "YAML::Tiny" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Dist::Zilla::Role::Plugin" => 0,
    "Dist::Zilla::Role::Releaser" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "File::Temp" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Path::Class" => 0,
    "Test::DZil" => 0,
    "Test::File::ShareDir" => 0,
    "Test::More" => "0.96",
    "lib" => 0
  },
  "VERSION" => "4.203",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Data::Section" => 0,
  "Dist::Zilla" => 5,
  "Dist::Zilla::Plugin::Authority" => "1.005",
  "Dist::Zilla::Plugin::AutoMetaResources" => 0,
  "Dist::Zilla::Plugin::AutoPrereqs" => 0,
  "Dist::Zilla::Plugin::Bootstrap::lib" => 0,
  "Dist::Zilla::Plugin::CheckChangesHasContent" => "0.006",
  "Dist::Zilla::Plugin::CheckExtraTests" => 0,
  "Dist::Zilla::Plugin::CheckMetaResources" => 0,
  "Dist::Zilla::Plugin::CheckPrereqsIndexed" => 0,
  "Dist::Zilla::Plugin::ConfirmRelease" => 0,
  "Dist::Zilla::Plugin::ContributorsFromGit" => "0.005",
  "Dist::Zilla::Plugin::ExecDir" => 0,
  "Dist::Zilla::Plugin::GatherDir" => 0,
  "Dist::Zilla::Plugin::GenerateFile" => 0,
  "Dist::Zilla::Plugin::Git::Describe" => 0,
  "Dist::Zilla::Plugin::Git::Init" => 0,
  "Dist::Zilla::Plugin::Git::NextVersion" => 0,
  "Dist::Zilla::Plugin::GithubMeta" => "0.10",
  "Dist::Zilla::Plugin::InstallRelease" => "0.006",
  "Dist::Zilla::Plugin::License" => 0,
  "Dist::Zilla::Plugin::MakeMaker" => 0,
  "Dist::Zilla::Plugin::Manifest" => 0,
  "Dist::Zilla::Plugin::ManifestSkip" => 0,
  "Dist::Zilla::Plugin::MetaConfig" => 0,
  "Dist::Zilla::Plugin::MetaJSON" => 0,
  "Dist::Zilla::Plugin::MetaNoIndex" => "1.10113",
  "Dist::Zilla::Plugin::MetaProvides::Package" => "1.14000001",
  "Dist::Zilla::Plugin::MetaYAML" => 0,
  "Dist::Zilla::Plugin::MinimumPerl" => "1.003",
  "Dist::Zilla::Plugin::NextRelease" => 0,
  "Dist::Zilla::Plugin::OurPkgVersion" => "0.002",
  "Dist::Zilla::Plugin::PkgVersion" => 0,
  "Dist::Zilla::Plugin::PodWeaver" => 0,
  "Dist::Zilla::Plugin::Prepender" => "1.112280",
  "Dist::Zilla::Plugin::PruneCruft" => 0,
  "Dist::Zilla::Plugin::PruneFiles" => 0,
  "Dist::Zilla::Plugin::Readme" => 0,
  "Dist::Zilla::Plugin::ReadmeAnyFromPod" => "0.120120",
  "Dist::Zilla::Plugin::Run" => "0.014",
  "Dist::Zilla::Plugin::Run::AfterMint" => 0,
  "Dist::Zilla::Plugin::ShareDir" => 0,
  "Dist::Zilla::Plugin::TemplateModule" => 0,
  "Dist::Zilla::Plugin::Test::ChangesHasContent" => "0.006",
  "Dist::Zilla::Plugin::Test::PodSpelling" => 0,
  "Dist::Zilla::Plugin::Test::ReportPrereqs" => "0.004",
  "Dist::Zilla::Plugin::TestRelease" => 0,
  "Dist::Zilla::Plugin::UploadToCPAN" => 0,
  "Dist::Zilla::PluginBundle::Git" => "2.004",
  "Dist::Zilla::PluginBundle::TestingMania" => "0.22",
  "Dist::Zilla::Role::MintingProfile::ShareDir" => 0,
  "Dist::Zilla::Role::Plugin" => 0,
  "Dist::Zilla::Role::PluginBundle::Config::Slicer" => 0,
  "Dist::Zilla::Role::PluginBundle::Easy" => 0,
  "Dist::Zilla::Role::PluginBundle::PluginRemover" => 0,
  "Dist::Zilla::Role::Releaser" => 0,
  "Dist::Zilla::Stash::PodWeaver" => "1.005",
  "ExtUtils::MakeMaker" => 0,
  "File::ShareDir::Install" => "0.06",
  "File::Spec" => 0,
  "File::Temp" => 0,
  "Git::Wrapper" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "List::Util" => 0,
  "Module::Build" => "0.3601",
  "Moose" => 0,
  "Moose::Util::TypeConstraints" => "1.01",
  "MooseX::AttributeShortcuts" => 0,
  "Path::Class" => 0,
  "Pod::Elemental::Transformer::List" => 0,
  "Pod::Markdown" => "2.000",
  "Pod::Weaver" => 4,
  "Pod::Weaver::Config::Assembler" => 0,
  "Pod::Weaver::Plugin::SingleEncoding" => 0,
  "Pod::Weaver::Plugin::StopWords" => "1.005",
  "Pod::Weaver::Plugin::Transformer" => 0,
  "Pod::Weaver::Plugin::WikiDoc" => 0,
  "Pod::Weaver::PluginBundle::CorePrep" => 0,
  "Pod::Weaver::Section::Authors" => 0,
  "Pod::Weaver::Section::Collect" => 0,
  "Pod::Weaver::Section::Contributors" => 0,
  "Pod::Weaver::Section::Generic" => 0,
  "Pod::Weaver::Section::Leftovers" => 0,
  "Pod::Weaver::Section::Legal" => 0,
  "Pod::Weaver::Section::Name" => 0,
  "Pod::Weaver::Section::Region" => 0,
  "Pod::Weaver::Section::Support" => "1.005",
  "Pod::Weaver::Section::Version" => 0,
  "Test::DZil" => 0,
  "Test::File::ShareDir" => 0,
  "Test::More" => "0.96",
  "YAML::Tiny" => 0,
  "lib" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}




lib/Pod/Weaver/PluginBundle/Author/RWSTAUNER.pm
# vim: set ts=2 sts=2 sw=2 expandtab smarttab:
#
# This file is part of Dist-Zilla-PluginBundle-Author-RWSTAUNER
#
# This software is copyright (c) 2010 by Randy Stauner.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#
use strict;
use warnings;

package Pod::Weaver::PluginBundle::Author::RWSTAUNER;
our $AUTHORITY = 'cpan:RWSTAUNER';
$Pod::Weaver::PluginBundle::Author::RWSTAUNER::VERSION = '4.203';
# ABSTRACT: RWSTAUNER's Pod::Weaver config

use Pod::Weaver::Config::Assembler;
sub _exp { Pod::Weaver::Config::Assembler->expand_package($_[0]) }

sub _plain {
  my ($plug, $arg) = (@_, {});
  (my $name = $plug) =~ s/^\W//;
  return [ $name, _exp($plug), { %$arg } ];
}

sub _bundle_name {
  my $class = @_ ? ref $_[0] || $_[0] : __PACKAGE__;
  join('', '@', ($class =~ /^.+::PluginBundle::(.+)$/));
}

sub mvp_bundle_config {
  ## ($self, $bundle) = @_; $bundle => {payload => {}, name => '@...'}
  my ($self) = @_;
  my @plugins;

  # NOTE: bundle name gets prepended to each plugin name at the end

  push @plugins, (
    # plugin
    _plain('-SingleEncoding'),
    _plain('-WikiDoc'),
    # default
    _plain('@CorePrep'),

    # sections
    # default
    _plain('Name'),
    _plain('Version'),

    # Any pod inside a =begin/end :prelude will go at the top
    [ 'Prelude',     _exp('Region'),  { region_name => 'prelude' } ],
  );

  for my $plugin (
    # default
    [ 'Synopsis',    _exp('Generic'), {} ],
    [ 'Description', _exp('Generic'), {} ],
    [ 'Overview',    _exp('Generic'), {} ],
    # extra
    [ 'Usage',       _exp('Generic'), {} ],

    ['Class Methods',_exp('Collect'), { command => 'class_method' } ], # header => 'CLASS METHODS',
    # default
    [ 'Attributes',  _exp('Collect'), { command => 'attr'   } ],
    [ 'Methods',     _exp('Collect'), { command => 'method' } ],
    [ 'Functions',   _exp('Collect'), { command => 'func'   } ],
  ) {
    $plugin->[2]{header} = uc $plugin->[0];
    push @plugins, $plugin;
  }

  # default
  push @plugins, (
    _plain('Leftovers'),
    # see prelude above
    [ 'Postlude',    _exp('Region'),    { region_name => 'postlude' } ],

    # TODO: consider SeeAlso if it ever allows comments with the links

    # extra
    # include Support section with various cpan links and github repo
    [ 'Support',     _exp('Support'),
      {
        ':version' => '1.005', # metacpan
        repository_content => '',
        repository_link => 'both',
        # metacpan links to everything else
        websites => [ qw(metacpan) ],
      }
    ],

    [ 'Acknowledgements', _exp('Generic'), {header => 'ACKNOWLEDGEMENTS'} ],

    _plain('Authors'),
    _plain('Contributors'),

    _plain('Legal'),

    # plugins
    [ 'List',        _exp('-Transformer'), { 'transformer' => 'List' } ],

    _plain('-StopWords', {
      ':version' => '1.005', # after =encoding
      # my dictionary doesn't like that extra 'E' but it looks funny without it
      include => 'ACKNOWLEDGEMENTS'
    }),
  );

  # prepend bundle name to each plugin name
  my $name = $self->_bundle_name;
  @plugins = map { $_->[0] = "$name/$_->[0]"; $_ } @plugins;

  return @plugins;
}

1;

__END__

=pod

=encoding UTF-8

=for :stopwords Randy Stauner ACKNOWLEDGEMENTS RWSTAUNER's PluginBundle Romanov Sergey

=head1 NAME

Pod::Weaver::PluginBundle::Author::RWSTAUNER - RWSTAUNER's Pod::Weaver config

=head1 VERSION

version 4.203

=head1 SYNOPSIS

  ; weaver.ini

  [@Author::RWSTAUNER]

or with a F<dist.ini> like so:

  ; dist.ini

  [@Author::RWSTAUNER]

you don't need a F<weaver.ini> at all.

=for Pod::Coverage mvp_bundle_config

=head1 ROUGHLY EQUIVALENT

This bundle is roughly equivalent to:

  [-SingleEncoding]
  [-WikiDoc]
  [@CorePrep]
  [Name]
  [Version]

  [Region / Prelude]
  region_name = prelude

  [Generic / Synopsis]
  header = SYNOPSIS

  [Generic / Description]
  header = DESCRIPTION

  [Generic / Overview]
  header = OVERVIEW

  [Generic / Usage]
  header = USAGE

  [Collect / Class Methods]
  command = class_method
  header  = CLASS METHODS

  [Collect / Attributes]
  command = attr
  header  = ATTRIBUTES

  [Collect / Methods]
  command = method
  header  = METHODS

  [Collect / Functions]
  command = func
  header  = FUNCTIONS

  [Leftovers]

  [Region / Postlude]
  region_name = postlude

  [Support]
  :version           = 1.005
  repository_content =
  repository_link    = both
  websites           = metacpan

  [Generic / Acknowledgements]
  header = ACKNOWLEDGEMENTS

  [Authors]
  [Contributors]
  [Legal]

  [-Transformer / List]
  transformer = List

  [-StopWords]
  :version = 1.005
  include  = ACKNOWLEDGEMENTS

=head1 AUTHOR

Randy Stauner <rwstauner@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2010 by Randy Stauner.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Build.PL
#
# This file is part of Dist-Zilla-PluginBundle-Author-RWSTAUNER
#
# This software is copyright (c) 2010 by Randy Stauner.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.025.
use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.3601"
  },
  "configure_requires" => {
    "File::ShareDir::Install" => "0.06",
    "Module::Build" => "0.3601",
    "perl" => "5.006"
  },
  "dist_abstract" => "RWSTAUNER's Dist::Zilla config",
  "dist_author" => [
    "Randy Stauner <rwstauner\@cpan.org>"
  ],
  "dist_name" => "Dist-Zilla-PluginBundle-Author-RWSTAUNER",
  "dist_version" => "4.203",
  "license" => "perl",
  "module_name" => "Dist::Zilla::PluginBundle::Author::RWSTAUNER",
  "recommends" => {
    "Devel::Cover" => "1.00",
    "Dist::Zilla::App::Command::cover" => 0,
    "Dist::Zilla::Plugin::DualBuilders" => "1.001",
    "Dist::Zilla::Plugin::TaskWeaver" => "0.101620",
    "Dist::Zilla::Plugin::Test::PodSpelling" => 2
  },
  "recursive_test_files" => 1,
  "requires" => {
    "Data::Section" => 0,
    "Dist::Zilla" => 5,
    "Dist::Zilla::Plugin::Authority" => "1.005",
    "Dist::Zilla::Plugin::AutoMetaResources" => 0,
    "Dist::Zilla::Plugin::AutoPrereqs" => 0,
    "Dist::Zilla::Plugin::Bootstrap::lib" => 0,
    "Dist::Zilla::Plugin::CheckChangesHasContent" => "0.006",
    "Dist::Zilla::Plugin::CheckExtraTests" => 0,
    "Dist::Zilla::Plugin::CheckMetaResources" => 0,
    "Dist::Zilla::Plugin::CheckPrereqsIndexed" => 0,
    "Dist::Zilla::Plugin::ConfirmRelease" => 0,
    "Dist::Zilla::Plugin::ContributorsFromGit" => "0.005",
    "Dist::Zilla::Plugin::ExecDir" => 0,
    "Dist::Zilla::Plugin::GatherDir" => 0,
    "Dist::Zilla::Plugin::GenerateFile" => 0,
    "Dist::Zilla::Plugin::Git::Describe" => 0,
    "Dist::Zilla::Plugin::Git::Init" => 0,
    "Dist::Zilla::Plugin::Git::NextVersion" => 0,
    "Dist::Zilla::Plugin::GithubMeta" => "0.10",
    "Dist::Zilla::Plugin::InstallRelease" => "0.006",
    "Dist::Zilla::Plugin::License" => 0,
    "Dist::Zilla::Plugin::MakeMaker" => 0,
    "Dist::Zilla::Plugin::Manifest" => 0,
    "Dist::Zilla::Plugin::ManifestSkip" => 0,
    "Dist::Zilla::Plugin::MetaConfig" => 0,
    "Dist::Zilla::Plugin::MetaJSON" => 0,
    "Dist::Zilla::Plugin::MetaNoIndex" => "1.10113",
    "Dist::Zilla::Plugin::MetaProvides::Package" => "1.14000001",
    "Dist::Zilla::Plugin::MetaYAML" => 0,
    "Dist::Zilla::Plugin::MinimumPerl" => "1.003",
    "Dist::Zilla::Plugin::NextRelease" => 0,
    "Dist::Zilla::Plugin::OurPkgVersion" => "0.002",
    "Dist::Zilla::Plugin::PkgVersion" => 0,
    "Dist::Zilla::Plugin::PodWeaver" => 0,
    "Dist::Zilla::Plugin::Prepender" => "1.112280",
    "Dist::Zilla::Plugin::PruneCruft" => 0,
    "Dist::Zilla::Plugin::PruneFiles" => 0,
    "Dist::Zilla::Plugin::Readme" => 0,
    "Dist::Zilla::Plugin::ReadmeAnyFromPod" => "0.120120",
    "Dist::Zilla::Plugin::Run" => "0.014",
    "Dist::Zilla::Plugin::Run::AfterMint" => 0,
    "Dist::Zilla::Plugin::ShareDir" => 0,
    "Dist::Zilla::Plugin::TemplateModule" => 0,
    "Dist::Zilla::Plugin::Test::ChangesHasContent" => "0.006",
    "Dist::Zilla::Plugin::Test::PodSpelling" => 0,
    "Dist::Zilla::Plugin::Test::ReportPrereqs" => "0.004",
    "Dist::Zilla::Plugin::TestRelease" => 0,
    "Dist::Zilla::Plugin::UploadToCPAN" => 0,
    "Dist::Zilla::PluginBundle::Git" => "2.004",
    "Dist::Zilla::PluginBundle::TestingMania" => "0.22",
    "Dist::Zilla::Role::MintingProfile::ShareDir" => 0,
    "Dist::Zilla::Role::PluginBundle::Config::Slicer" => 0,
    "Dist::Zilla::Role::PluginBundle::Easy" => 0,
    "Dist::Zilla::Role::PluginBundle::PluginRemover" => 0,
    "Dist::Zilla::Stash::PodWeaver" => "1.005",
    "Git::Wrapper" => 0,
    "List::Util" => 0,
    "Moose" => 0,
    "Moose::Util::TypeConstraints" => "1.01",
    "MooseX::AttributeShortcuts" => 0,
    "Pod::Elemental::Transformer::List" => 0,
    "Pod::Markdown" => "2.000",
    "Pod::Weaver" => 4,
    "Pod::Weaver::Config::Assembler" => 0,
    "Pod::Weaver::Plugin::SingleEncoding" => 0,
    "Pod::Weaver::Plugin::StopWords" => "1.005",
    "Pod::Weaver::Plugin::Transformer" => 0,
    "Pod::Weaver::Plugin::WikiDoc" => 0,
    "Pod::Weaver::PluginBundle::CorePrep" => 0,
    "Pod::Weaver::Section::Authors" => 0,
    "Pod::Weaver::Section::Collect" => 0,
    "Pod::Weaver::Section::Contributors" => 0,
    "Pod::Weaver::Section::Generic" => 0,
    "Pod::Weaver::Section::Leftovers" => 0,
    "Pod::Weaver::Section::Legal" => 0,
    "Pod::Weaver::Section::Name" => 0,
    "Pod::Weaver::Section::Region" => 0,
    "Pod::Weaver::Section::Support" => "1.005",
    "Pod::Weaver::Section::Version" => 0,
    "YAML::Tiny" => 0,
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "share_dir" => {
    "dist" => "share",
    "module" => {
      "Dist::Zilla::MintingProfile::Author::RWSTAUNER" => "share/profiles"
    }
  },
  "test_requires" => {
    "Dist::Zilla::Role::Plugin" => 0,
    "Dist::Zilla::Role::Releaser" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "File::Temp" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Path::Class" => 0,
    "Test::DZil" => 0,
    "Test::File::ShareDir" => 0,
    "Test::More" => "0.96",
    "lib" => 0,
    "perl" => "5.006"
  }
);


my %fallback_build_requires = (
  "Dist::Zilla::Role::Plugin" => 0,
  "Dist::Zilla::Role::Releaser" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::Temp" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Module::Build" => "0.3601",
  "Path::Class" => 0,
  "Test::DZil" => 0,
  "Test::File::ShareDir" => 0,
  "Test::More" => "0.96",
  "lib" => 0,
  "perl" => "5.006"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;
